packages/camera/android/src/main/java/io/flutter/plugins/camera/CameraPlugin.java

```diff
@@ -4,172 +4,107 @@
-    switch (call.method) {
-      case "availableCameras":
-        try {
-          result.success(CameraUtils.getAvailableCameras(registrar.activity()));
-        } catch (Exception e) {
-          handleException(e, result);
-        }
-        break;
-      case "initialize":
-        {
-          if (camera != null) {
-            camera.close();
-          }
-          cameraPermissions.requestPermissions(
-              registrar,
-              call.argument("enableAudio"),
-              (String errCode, String errDesc) -> {
-                if (errCode == null) {
-                  try {
-                    instantiateCamera(call, result);
-                  } catch (Exception e) {
-                    handleException(e, result);
-                  }
-                } else {
-                  result.error(errCode, errDesc, null);
-                }
-              });
+  public void onAttachedToActivity(@NonNull ActivityPluginBinding binding) {
+    startListening(
+        binding.getActivity(),
+        flutterPluginBinding.getFlutterEngine().getDartExecutor(),
+        binding::addRequestPermissionsResultListener,
+        flutterPluginBinding.getFlutterEngine().getRenderer());
+  }
 
-          break;
-        }
-      case "takePicture":
-        {
-          camera.takePicture(call.argument("path"), result);
-          break;
-        }
-      case "prepareForVideoRecording":
-        {
-          // This optimization is not required for Android.
-          result.success(null);
-          break;
-        }
-      case "startVideoRecording":
-        {
-          camera.startVideoRecording(call.argument("filePath"), result);
-          break;
-        }
-      case "stopVideoRecording":
-        {
-          camera.stopVideoRecording(result);
-          break;
-        }
-      case "pauseVideoRecording":
-        {
-          camera.pauseVideoRecording(result);
-          break;
-        }
-      case "resumeVideoRecording":
-        {
-          camera.resumeVideoRecording(result);
-          break;
-        }
-      case "startImageStream":
-        {
-          try {
-            camera.startPreviewWithImageStream(imageStreamChannel);
-            result.success(null);
-          } catch (Exception e) {
-            handleException(e, result);
-          }
-          break;
-        }
-      case "stopImageStream":
-        {
-          try {
-            camera.startPreview();
-            result.success(null);
-          } catch (Exception e) {
-            handleException(e, result);
-          }
-          break;
-        }
-      case "dispose":
-        {
-          if (camera != null) {
-            camera.dispose();
-          }
-          result.success(null);
-          break;
-        }
-      default:
-        result.notImplemented();
-        break;
+  @Override
+  public void onDetachedFromActivity() {
+    if (methodCallHandler == null) {
+      Log.wtf(TAG, "Detached before initialzed.");
```

Comment:
offline discussion:
This is not necessary if assuming `onDetachedFromActivity` is always called after `onAttachedToActivity`
We leave the check here just in case since the v2 embedding is untested.